force-app/main/default/classes/Addresses.cls

```diff
@@ -485,69 +569,69 @@ public inherited sharing class Addresses extends fflib_SObjects {
         return listAddrS;
     }
 
-    public Map<Id, Address__c> addressChangesToSync() {
-        Map<Id, Address__c> defaultAndSeasonalWithChanges = new Map<Id, Address__c>();
-        Integer i = 0;
-        for (Address__c addr : (List<Address__c>) getRecords()) {
-            NPSP_Address npspAddress = new NPSP_Address(addr);
-            Address__c addrOld = oldVersionOf(addr);
-            NPSP_Address npspAddressOld = new NPSP_Address(addrOld);
-            // an address that is marked default needs to propagate to the household
-            if (isNewDefault(addr, addrOld) ||
-                isSeasonalChanged(addr, addrOld) ||
-                (isUndeliverableStatusChanged(addr, addrOld)) && isDefault(addr)) {
-
-                defaultAndSeasonalWithChanges.put(householdAccountIdFor(addr), addr);
-            } else if (!npspAddress.isEqualToIncludingAddressType(npspAddressOld, TRUE)) {
-                if (isDefault(addr)) {
-                    defaultAndSeasonalWithChanges.put(householdAccountIdFor(addr), addr);
-                }
+    public Map<Id, Address__c> addressChangesToSyncToHouseholdAccountsAndContacts() {
+        Map<Id, Address__c> addressesByAccountId = new Map<Id, Address__c>();
+
+        for (Address__c address : (List<Address__c>) getRecords()) {
+            Address__c addressOld = oldVersionOf(address);
+
+            if (shouldSyncAddressToHouseholdAccountAndContacts(address, addressOld)) {
+                addressesByAccountId.put(householdAccountIdFor(address), address);
             }
-            i++;
         }
-        return defaultAndSeasonalWithChanges;
+        return addressesByAccountId;
     }
 
-    private Boolean isUndeliverableStatusChanged(Address__c addr, Address__c addrOld) {
-        return addr.Undeliverable__c != addrOld.Undeliverable__c;
-    }
+    private Boolean shouldSyncAddressToHouseholdAccountAndContacts(Address__c address, Address__c oldAddress) {
```

Comment:
I think most of this is also able to be an instance method on NPSP_Address as well. I see you've done a few refactors like this in this PR already, with functions that use the old/new version of an Address.  The `isSeasonalChanged(Address__c addr, Address__c oldAddress)` function would also get moved over to that Entity class.

Reply:
great callout. updated!